src/gen9_avc_encoder.c

```diff
@@ -0,0 +1,6031 @@
+            0,
+            0
+    },
+
+    // unsigned int 20
+    {
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 21
+    {
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 22
+    {
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 23
+    {
+            0
+    }
+};
+
+const gen9_avc_frame_brc_update_curbe_data gen9_avc_frame_brc_update_curbe_init_data =
+{
+    // unsigned int 0
+    {
+            0
+    },
+
+    // unsigned int 1
+    {
+            0
+    },
+
+    // unsigned int 2
+    {
+            0
+    },
+
+    // unsigned int 3
+    {
+            10,
+            50
+    },
+
+    // unsigned int 4
+    {
+            100,
+            150
+    },
+
+    // unsigned int 5
+    {
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 6
+    {
+            0,
+            0,
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 7
+    {
+            0
+    },
+
+    // unsigned int 8
+    {
+            1,
+            1,
+            3,
+            2
+    },
+
+    // unsigned int 9
+    {
+            1,
+            40,
+            5,
+            5
+    },
+
+    // unsigned int 10
+    {
+            3,
+            1,
+            7,
+            18
+    },
+
+    // unsigned int 11
+    {
+            25,
+            37,
+            40,
+            75
+    },
+
+    // unsigned int 12
+    {
+            97,
+            103,
+            125,
+            160
+    },
+
+    // unsigned int 13
+    {
+            -3,
+            -2,
+            -1,
+            0
+    },
+
+    // unsigned int 14
+    {
+            1,
+            2,
+            3,
+            0xff
+    },
+
+    // unsigned int 15
+    {
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 16
+    {
+            0
+    },
+
+    // unsigned int 17
+    {
+            0
+    },
+
+    // unsigned int 18
+    {
+            0
+    },
+
+    // unsigned int 19
+    {
+            0
+    },
+
+    // unsigned int 20
+    {
+            0
+    },
+
+    // unsigned int 21
+    {
+            0
+    },
+
+    // unsigned int 22
+    {
+            0
+    },
+
+    // unsigned int 23
+    {
+            0
+    },
+
+};
+
+static void
+gen9_avc_update_misc_parameters(VADriverContextP ctx,
+                                struct encode_state *encode_state,
+                                struct intel_encoder_context *encoder_context)
+{
+    struct encoder_vme_mfc_context * vme_context = (struct encoder_vme_mfc_context *)encoder_context->vme_context;
+    struct generic_enc_codec_state * generic_state = (struct generic_enc_codec_state * )vme_context->generic_enc_state;
+    int i;
+
+    /* brc */
+    generic_state->max_bit_rate = ALIGN(encoder_context->brc.bits_per_second[0], 1000) / 1000;
```

Comment:
@xhaihao You know this issue has caught people more than once.  In future we may need to explicitly define ALIGN to be something ALIGNPWR2 or something.